src/odb_loose.c

```diff
@@ -910,28 +911,29 @@ static int loose_backend__writestream(git_odb_stream **stream_out, git_odb_backe
 static int loose_backend__readstream_read(
 	git_odb_stream *_stream,
 	char *buffer,
-	size_t len)
+	size_t buffer_len)
 {
 	loose_readstream *stream = (loose_readstream *)_stream;
+	size_t start_remain = stream->start_len - stream->start_read;
 	int total = 0, error;
 
 	/*
 	 * if we read more than just the header in the initial read, play
 	 * that back for the caller.
 	 */
-	if (stream->start_len && len) {
-		size_t chunk = MIN(stream->start_len, len);
-		memcpy(buffer, stream->start, chunk);
+	if (start_remain && buffer_len) {
+		size_t chunk = min(start_remain, buffer_len);
+		memcpy(buffer, stream->start + stream->start_read, chunk);
```

Comment:
Ah, here it's getting fixed. Did you squash that into the wrong commit by accident?